src/main/ngapp/accio/src/app/editor/editor.component.ts

```diff
@@ -27,95 +37,161 @@ export class EditorComponent implements OnInit {
     let imgHeight = this.image.height;
 
-    // Initialize hidden canvas width and height to original images width and height.
+    //  Initialize hidden canvas width and height to original images width and height.
     this.hiddenCanvas.nativeElement.width = imgWidth;
     this.hiddenCanvas.nativeElement.height = imgHeight;
 
-    // Used to scale the image to the window size, @Param scaledFactor = .9 so the scaled image is smaller than the users window.
+    //  Used to scale the image to the window size, @Param scaledFactor = .9 so the scaled image is smaller than the users window.
     this.scaleFactor = Math.floor(this.innerHeight / imgHeight * this.scaleFactor);
-    // TODO(shmcaffrey): add scaling if image is larger than window
+    //  TODO(shmcaffrey): add scaling if image is larger than window
     if (this.scaleFactor <= 0) {
       this.scaleFactor =  1;
     }
 
-    // Adjust canvas to scaled image width and height, use ctx. 
+    //  Adjust canvas to scaled image width and height, use ctx. 
     this.canvas.nativeElement.width = imgWidth * this.scaleFactor;
     this.canvas.nativeElement.height = imgHeight * this.scaleFactor;
     this.ctx.scale(this.scaleFactor, this.scaleFactor); 
 
     //  Initalize transparent black image data to use for mask.
-    this.maskData = new ImageData(imgWidth,  imgHeight);
+    this.maskImageData = new ImageData(imgWidth,  imgHeight);
     
     this.image.onload = () => {
       this.ctx.drawImage(this.image, 0, 0, imgWidth, imgHeight);
       this.hiddenCtx.drawImage(this.image, 0, 0)
     }
   }
 
-  /** Returns the calculated scale for the image loaded. */
+  /**  Returns the calculated scale for the image loaded. */
   public getScaleFactor(): number {
     return this.scaleFactor;
   }
 
-  /** Returns the original images data for reference in mask making. */
-  public getImgData(): ImageData {
+  /**  Returns the original images data for reference in mask making. */
+  public getOriginalImageData(): ImageData {
     return this.hiddenCtx.getImageData(0,0, this.image.width, this.image.height);
   }
 
-  /** Returns black transparent ImageData for single mask image. */
-  public getMaskData(): ImageData {
-    return this.maskData;
+  /**  Returns black transparent ImageData for single mask image. */
+  public getMaskImageData(): ImageData {
+    return this.maskImageData;
   }
 
-  /** Returns the scaled images data for reference printing scaled image after mask. */
+  /**  Returns the scaled images data for reference printing scaled image after mask. */
   public getScaledData(): ImageData {
     return this.ctx.getImageData(0, 0, this.image.width * this.scaleFactor, this.image.height * this.scaleFactor);
   }
+
+  /** 
+   *  Clears canvas to get mask Image as url to store and redraws
+   *    original image for possibility user makes more edits. 
+   * @return Url for the mask image to be stored in blobstore.  
+   */
+  async getMaskBlob(): Promise<void> {
```

Comment:
Since you are using the hiddenCtx in both the getImgData function and the getMaskBlob function there is the potential for a race condition and conflict.  How are you avoiding that?  It would be good to clarify that in comments in the code.

Reply:
I noted this in the comments, I am going to re factor mask.directive and editor.ts today so I will account for this then.